simulation/modules/macrophage.py

```diff
@@ -0,0 +1,167 @@
+from simulation.modules.geometry import GeometryState, TissueTypes
+
+class MacrophageCellData(CellData):
+    BOOLEAN_NETWORK_LENGTH = 3 #place holder for now
+
+    class Status(IntEnum):
+        RESTING = 0
+        ACTIVE = 1
+        INTERACTING = 2
+        SECRETING = 3
+        SYNERGIC = 4
+        APOPTOTIC = 5
+        NECROTIC = 6
+        DEAD = 7
+
+    MACROPHAGE_FIELDS = [
+        ('boolean_network', 'b1', BOOLEAN_NETWORK_LENGTH),
+        ('status', 'u1'),
+        ('iron_pool', 'f8'),
+        ('iteration', 'i4'),
+    ]
+
+    dtype = np.dtype(CellData.FIELDS + MACROPHAGE_FIELDS, align=True)  # type: ignore
+
+    @classmethod
+    def create_cell_tuple(
+        cls,
+        *,
+        iron_pool: float = 0,
+        status: Status = Status.RESTING,
+        **kwargs,
+    ) -> np.record:
+
+        network = cls.initial_boolean_network()
+        iteration = 0
+
+        return CellData.create_cell_tuple(**kwargs) + (
+            network,
+            status,
+            iron_pool,
+            iteration,
+        )
+
+    @classmethod
+    def initial_boolean_network(cls) -> np.ndarray:
+        return np.asarray(
+            [
+                True,
+                False,
+                True,
+            ]
+        )
+
+
+@attr.s(kw_only=True, frozen=True, repr=False)
+class MacrophageCellList(CellList):
+    CellDataClass = MacrophageCellData
+
+    def is_moveable(self, grid: RectangularGrid):
+        cells = self.cell_data
+        return self.alive(
+            (cells['status'] == MacrophageCellData.Status.RESTING)
+            & cells.point_mask(cells['point'], grid)
+        )
+
+def cell_list_factory(self: 'MacrophageState'):
+    return MacrophageCellList(grid=self.global_state.grid)
+
+@attr.s(kw_only=True)
+class MacrophageState(ModuleState):
+    cells: MacrophageCellList = attr.ib(default=attr.Factory(cell_list_factory, takes_self=True))
+    init_num: int
+
+class Macrophage(Module):
+    name = 'macrophage'
+    defaults = {
+        'cells': '',
+        'init_num': '0',
+    }
+    StateClass = MacrophageState
+
+    def initialize(self, state: State):
+        macrophage: MacrophageState = state.macrophage
+        grid: RectangularGrid = state.grid
+        tissue: GeometryState = state.geometry.lung_tissue
+
+        macrophage.init_num = self.config.getint('init_num')
+        macrophage.cells = MacrophageCellList(grid=grid)
+
+        if(macrophage.init_num > 0):
+            #initialize the surfactant layer with some macrophage in random locations
+            indices = np.argwhere(tissue == TissueTypes.SURFACTANT.value)
+            np.random.shuffle(indices)
+
+            for i in range(0, macrophage.init_num):
+                x = indices[i][2] + (random.uniform(-0.5, 0.5))
+                y = indices[i][1] + (random.uniform(-0.5, 0.5))
+                z = indices[i][0] + (random.uniform(-0.5, 0.5))
+                
+                point = Point(x=x, y=y, z=z)

```

Comment:
The logic here isn't right.

The spatial coordinates of the voxel at `ix, iy, iz` are:
* `x` within the range `grid.xv[ix]` to `grid.xv[ix + 1]`
* `y` within the range `grid.yv[iy]` to `grid.yv[iy + 1]`
* `z` within the range `grid.zv[iz]` to `grid.zv[iz + 1]`